src/redux/uniqueTokens.ts

```diff
@@ -266,11 +269,35 @@ export const fetchUniqueTokens = (showcaseAddress?: string) => async (
   const fetchPage = async (page: number, network: Network) => {
     let shouldStopFetching = false;
     try {
-      const newPageResults = await apiGetAccountUniqueTokens(
+      let newPageResults = await apiGetAccountUniqueTokens(
         network,
         accountAddress,
         page
       );
+
+      // If there are any "unknown" ENS names, fallback to the ENS
+      // metadata service.
+      newPageResults = await Promise.all(
+        newPageResults.map(async (token: UniqueAsset) => {
+          const isENS =
+            token?.asset_contract?.address?.toLowerCase() ===
+            ENS_NFT_CONTRACT_ADDRESS.toLowerCase();
+          if (isENS && !token.uniqueId.includes('.eth')) {
```

Comment:
maybe adding a try catch here returning `token` if there's any error would be a good idea, if one of these fail it would break this flow

Reply:
yep sounds good!